server/src/test/java/io/crate/statistics/TransportAnalyzeActionTest.java

```diff
@@ -64,4 +103,68 @@ public void test_create_stats_for_tables_with_array_columns_with_nulls() {
         var stats = samples.createTableStats(references);
         assertThat(stats.numDocs()).isEqualTo(2L);
     }
+
+    @Test
+    public void test_existing_run_reused_even_if_table_schema_was_updated() throws Exception {
+        SQLExecutor executor = SQLExecutor.of(clusterService)
+            .addTable(
+                """
+                create table doc.t (
+                    a int
+                )
+                """
+            );
+        ReservoirSampler mockSampler = mock(ReservoirSampler.class);
+        when(mockSampler.getSamples(any(), anyList())).thenAnswer(_ -> {
+            // Sleep a bit to be sure that second request always arrives when first is still running.
+            Thread.sleep(100);
+            return Samples.EMPTY;
+        });
+
+        try (TransportService service = createTransportService()) {
+            TransportAnalyzeAction transportAnalyzeAction = new TransportAnalyzeAction(
+                service,
+                mockSampler,
+                executor.nodeCtx,
+                clusterService,
+                null // TableStatsService won't be used, is irrelevant for this test.
+            );
+
+            RelationName relationName = new RelationName(Schemas.DOC_SCHEMA_NAME, "t");
+            DocTableInfo table = new DocTableInfoFactory(executor.nodeCtx)
+                .create(relationName, clusterService.state().metadata());
+            Reference ref = table.getReference(ColumnIdent.of("a"));
+            assert ref != null : "ref must be resolved";
+            transportAnalyzeAction.fetchSamplesThenGenerateAndPublishStats();
+
+            transportAnalyzeAction.fetchSamplesThenGenerateAndPublishStats();
```

Comment:
This tests passes on master and fails on master if I imitate schema change between `fetchSamplesThenGenerateAndPublishStats` calls.

I didn't add schema change scenario as it's irrelevant for the new implementation - we just have 1 run, similar to auto scheduled thing. Otherwise would have to add quite some mocking and stuff like that only to cover something irrelevant.